apps/backend/src/datasources/stfc/StfcUserDataSource.ts

```diff
@@ -533,4 +533,23 @@ export class StfcUserDataSource implements UserDataSource {
       true
     ));
   }
+
+  async getUsersRoles(
+    userIds: number[]
+  ): Promise<{ userId: number; roles: Role[] }[]> {
+    const usersRolesPromises = [];
+    for (const userId of userIds) {
+      usersRolesPromises.push(this.getUserRoles(userId));
+    }
+
+    const usersRoles = await Promise.all(usersRolesPromises);
+
+    const usersWithRoles: { userId: number; roles: Role[] }[] = [];
+
+    usersRoles.forEach((roles, index) =>
```

Comment:
It might work to use map here with Promise.all

```typescript 
 const usersWithRoles: { userId: number; roles: Role[] }[] = await Promise.all(
    userIds.map(async (userId) => ({
      userId,
      roles: await this.getUserRoles(userId),
    })
  );
```